Drawer/Drawer/Drawer/DrawerModels.swift

```diff
@@ -19,49 +19,101 @@ public enum Drawer {
         case whenMinimised
     }
     
-    public enum EmbeddedAction {
-        case layoutUpdated(config: Drawer.ContentConfiguration)
+    public enum Action {
+        case layoutUpdated(config: Drawer.Configuration)
         case changeState(to: MovementState)
         
         public enum MovementState: Int {
-            case minimise
+            case minimize
             case fullScreen
             case dismiss
         }
     }
 
-    public struct ContentConfiguration {
+    public struct Configuration {
+
         let duration: TimeInterval
         let embeddedFullHeight: CGFloat
         let embeddedMinimumHeight: CGFloat
         let state: Drawer.State
-        let animateStateChange: Bool
-        let cornerRadius: ContentConfiguration.CornerRadius
+        let cornerRadius: Configuration.CornerRadius
         let dismissCompleteCallback: (() -> Void)?
-        
-        public init(duration: TimeInterval,
-                    embeddedFullHeight: CGFloat,
-                    state: Drawer.State,
-                    animateStateChange: Bool,
-                    embeddedMinimumHeight: CGFloat,
-                    cornerRadius: ContentConfiguration.CornerRadius? = nil,
+
+        /**
+         Creates a Configuration for the Drawer
+
+         - Parameter options: Use options to configure the Drawer.
+         - Parameter dismissCompleteCallback: Called when the Drawer has finished dismissing
+
+         Available configurations:
+         - .animationDuration: TimeInterval
+         -- default value is 0.3
+
+         - .fullHeight: CGFloat
+         -- default value is 300
+
+         - .minimunHeight: CGFloat
+         -- default value is 100
+
+         - .initialState: Drawer.State
+         -- default value is .minimized
+
+         - .cornerRadius: ContentConfiguration.CornerRadius
+         -- default value is CornerRadius(fullSize: 0, minimized: 0)
+
+         */
+        public init(options: [Drawer.Configuration.Key: Any],
                     dismissCompleteCallback: (() -> Void)? = nil) {
-            self.duration = duration
-            self.embeddedMinimumHeight = embeddedMinimumHeight
-            self.embeddedFullHeight = embeddedFullHeight
-            self.state = state
-            self.animateStateChange = animateStateChange
-            self.cornerRadius = cornerRadius ?? CornerRadius(fullSize: 0, minimised: 0)
+
+            if let duration = options[.animationDuration] as? TimeInterval {
+                self.duration = duration
+            } else {
+                self.duration = 0.3
+            }
```

Comment:
Couldn't all of these be rewritten as:

`self.duration = options[.animationDuration] as? TimeInterval ?? 0.3`
`self.embeddedFullHeight = options[.fullHeight] as? CGFloat ?? 300`

And so on?

Reduces the amount of code, always a good thing 😄